### ISSUE TITLE
[BUG] 2.2 + Request Caching and HTML Snippets Play Badly

### ISSUE THREAD

**Issue by rossriley** [bolt/bolt#4988]
Details
-------

It seems that the block of code in the after handler that auto-adds the meta generator tag and the canonical tag will still keep adding extra tags to the head of a cached request.

Reproduction
------------

This section is fittingly titled...

1. In config.yml turn on request caching.
2. Ensure that htmlsnippets are turned on
3. Make sure you are logged out of Bolt since authenticated requests aren't cached.
4. Watch the wild reproudction of meta tags of which an extra set get added on each request.

**Comment by bobdenotter** [comment]
Yes, This is something i've noticed too, a while ago. It's fixed for `master`, but not for 2.x. Do you reckon it's important enough to backport the fix. From what I recall, it wasn't trivial, but that could also have been my lack of overview.

**Comment by rossriley** [comment]
Well it's a fairly serious bug if it happens to you since the HTML docuement keeps growing and growing until it eventually starts crashing the browser.

There's a simple way to fix it and that would be to modify this line in the `afterHandler`
```
        if (isset($this['htmlsnippets']) && ($this['htmlsnippets'] === true)) {
```
to also check to see if request caching is turned on and not add the tags.

**Comment by GawainLynch** [comment]
> There's a simple way to fix it and that would be to modify this line in the afterHandler

:+1: 

The stuff from master would be a massive job to back port, tbh.
